src/services/bsc-scan.service.ts

```diff
@@ -0,0 +1,91 @@
+import fetch from 'node-fetch';
+import { BscScanData } from '../models/bsc-scan.response'
+
+const burnAddressesList: string[] = [
+    '0x000000000000000000000000000000000000dead',
+    '0x0000000000000000000000000000000000000000',
+    '0x0000000000000000000000000000000000000001',
+    '0x0000000000000000000000000000000000000005',
+    '0x0000000000000000000000000000000000000003',
+    '0x0000000000000000000000000000000000000004',
+    '0x0000000000000000000000000000000000000002',
+    '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee',
+    '0x1111111111111111111111111111111111111111',
+    '0xdead000000000000000042069420694206942069',
+    '0xbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb',
+    '0x6666666666666666666666666666666666666666',
+    '0x3333333333333333333333333333333333333333',
+    '0x64b00b5ec6df675e94736bdcc006dbd9a0b8b00b',
+    '0x8888888888888888888888888888888888888888',
+    '0x0000000000000000000000000000000000000008',
+    '0x2222222222222222222222222222222222222222',
+    '0xffffffffffffffffffffffffffffffffffffffff',
+    '0x0000000000000000000000000000000000000007',
+    '0x0000000000000000000000000000000000000006',
+    '0x0000000000000000000000000000000000000009',
+    '0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'
+];
+
+export async function getTokenTotalSupply(contractAddress: string): Promise<BscScanData> {
+    const url = "https://api.bscscan.com/api" +
+        "?module=stats" +
+        "&action=tokensupply" +
+        `&contractaddress=${contractAddress}` +
+        `&apikey=${process.env.BSC_API_KEY}`
+
+    return await fetch(url)
+        .then(res => res.json())
+        .then(json => new BscScanData(json));
+};
+
+export async function getCirculatingSupply(contractAddress: string): Promise<BscScanData> {
+    const url = "https://api.bscscan.com/api" +
+        "?module=stats" +
+        "&action=tokenCsupply" +
```

Comment:
Let's use this formula to get the circulating supply --> **getTokenTotalSupply - getBurnedTokenAmount** here as the bscscan API doesn't calculate the burned tokens properly..

Reply:
Good point, I did not notice that.